ipywidgets/widgets/widget_upload.py

```diff
@@ -16,12 +17,107 @@
 from .widget_button import ButtonStyle
 from .widget import register, widget_serialization
 from .trait_types import InstanceDict
+from traitlets.utils.bunch import Bunch
+
+
+def _deserialize_single_file(js):
+    uploaded_file = Bunch()
+    for attribute in ['name', 'type', 'size', 'content']:
+        uploaded_file[attribute] = js[attribute]
+    uploaded_file['last_modified'] = dt.datetime.fromtimestamp(
+        js['lastModified'] / 1000,
+        tz=dt.timezone.utc
+    )
+    return uploaded_file
+
+
+def _deserialize_value(js, _):
+    return [_deserialize_single_file(entry) for entry in js]
+
+
+def _serialize_single_file(uploaded_file):
+    js = {}
+    for attribute in ['name', 'type', 'size', 'content']:
+        js[attribute] = uploaded_file[attribute]
+    js['lastModified'] = int(uploaded_file['last_modified'].timestamp() * 1000)
```

Comment:
Our general convention, at least in default controls, has been snake_case for the attributes in the protocol messages. Should we keep it snake_case here, as `last_modified`?

Reply:
Another option is to just call it `modified`, or even `mtime`, which mirrors the `stat()` structure and how you address it in python and presumably lots of other languages that let you make a stat call.

Reply:
`mtime` sounds good.